### DISCUSSION TITLE
Clarification on adding RHEL8/9 channels in Uyuni: Is this supported?

### DISCUSSION THREAD

**Discussion by johanneskastl** [uyuni-project/uyuni#7232] [Q&A] [answered]
## Clarification wanted

I asked this on the mailinglist a while back, and Zaher gladly answered. But the point of whether this is officially supported or whether this needs a SCC subscription was not clarified. Hence starting this discussion to hopefully get an answer to that...

Zaher apparently added the channels manually:
https://github.com/uyuni-project/uyuni/issues/6566
https://github.com/uyuni-project/uyuni/issues/6563

## Problem description

I tried to add RHEL9 according to this documentation:
https://www.uyuni-project.org/uyuni-docs/en/uyuni/client-configuration/clients-rh-cdn.html

I uploaded the three files using the exact names from the documenation (not sure if the "date" in Entitlement-Cert-date should be replaced by a date?).

Anyhow, trying to add those channels simply spits out "No channels matching your selection".

```
spacewalk-common-channels \
el9-pool-x86_64 \
el9-manager-tools-pool-x86_64 \
el9-manager-tools-updates-x86_64
```

Searching for channels using "spacewalk-common-channels --list" does not find them:

```
# spacewalk-common-channels --list|grep -i rhel
# spacewalk-common-channels --list|grep -i el9
 almalinux9-epel9:    x86_64, aarch64, ppc64le, s390x
 rockylinux9-epel9:   x86_64, aarch64, ppc64le, s390x
# spacewalk-common-channels --list|grep -i el8
 almalinux8-epel8:    x86_64, ppc64le, aarch64
 epel8:               x86_64, ppc64le, aarch64
 rockylinux8-epel8:   x86_64, ppc64le, aarch64
# spacewalk-common-channels --list|grep -i el7
 epel7:   #
```

Server version was 2023.03. RHEL8 showed the same behaviour.

**Comment by mcalmer** [comment]
The docu is wrong. The names are the names for SUSE Manager when you have SCC credentials with a SUSE Manager product.
For Uyuni you should use:

```
spacewalk-common-channels -a x86_64 rhel9-pool-uyuni rhel9-uyuni-client
```

(-a x86_64 if you want only this architecture)

Feel free top open a bug issue or directly a PR against our docs.

**Reply by johanneskastl** [reply]
Thanks Michael for the clarification. I'll test this and create a PR for the docs...

**Reply by johanneskastl** [reply]
Hmmm, I tested this on a freshly installed Uyuni 2023.04 server, directly after the installation.

```bash
uyuniserver01:~ # spacewalk-common-channels -a x86_64 rhel9-pool-uyuni rhel9-uyuni-client
SUSE Manager username: admin
SUSE Manager password: 
No channels matching your selection.
uyuniserver01:~ #
```

Is this because no RHEL-secrets (certificates) were uploaded yet? Merely asking because `spacewalk-common-channels --list|grep -i rhel9` does not give any output...

I will test this with the actual server that has the secrets uploaded next week and report back.

**Reply by mcalmer** [reply, marked as answer]
Ahh, sorry. This is a fix in Master which is not yet released. It will be part of the next uyuni release.
You can fetch the config from git (https://raw.githubusercontent.com/uyuni-project/uyuni/master/utils/spacewalk-common-channels.ini) and put it at /etc/rhn/ . Then it should work.

**Reply by johanneskastl** [reply]
Thanks Michael, that worked (even without the secrets/credentials, I guess the sync will fail due to this, but the command itself worked.
